internal/setup/setup.go

```diff
@@ -318,20 +316,36 @@ func readSlices(release *Release, baseDir, dirName string) error {
 	return nil
 }
 
-type yamlRelease struct {
-	Format   string                 `yaml:"format"`
-	Archives map[string]yamlArchive `yaml:"archives"`
-	PubKeys  map[string]yamlPubKey  `yaml:"v1-public-keys"`
+var ubuntuAdjectives = map[string]string{
+	"18.04": "bionic",
+	"20.04": "focal",
+	"22.04": "jammy",
+	"22.10": "kinetic",
 }
 
-const yamlReleaseFormat = "chisel-v1"
+func parseRelease(baseDir, filePath string, data []byte) (*Release, error) {
+	type yamlFormat struct {
+		Format string `yaml:"format"`
+	}
+
+	fileName := stripBase(baseDir, filePath)
+
+	yamlVar := yamlFormat{}
+	dec := yaml.NewDecoder(bytes.NewBuffer(data))
+	dec.KnownFields(false)
+	err := dec.Decode(&yamlVar)
+	if err != nil {
+		return nil, fmt.Errorf("%s: cannot parse release definition: %v", fileName, err)
+	}
 
-type yamlArchive struct {
-	Version    string   `yaml:"version"`
-	Suites     []string `yaml:"suites"`
-	Components []string `yaml:"components"`
-	Default    bool     `yaml:"default"`
-	PubKeys    []string `yaml:"v1-public-keys"`
+	switch yamlVar.Format {
```

Comment:
[For reviewer]: I have chosen to add the two formats separately because that is the more flexible option. If we end up having to support `chisel-v1` longer than expected or we need to do some change in `v1`, it would be far easier to do it if they are not entangled.

The implementation happens to be similar but they represent different format so, in my opinion, we should not strive for DRY here.